src/physics/exact_solutions/exact_solution.cpp

```diff
@@ -0,0 +1,95 @@
+#include <deal.II/base/function.h>
+#include "exact_solution.h"
+
+namespace PHiLiP {
+
+// ========================================================
+// ZERO -- Returns zero everywhere; used a placeholder when no exact solution is defined.
+// ========================================================
+template <int dim, int nstate, typename real>
+ExactSolutionFunction_Zero<dim,nstate,real>
+::ExactSolutionFunction_Zero()
+        : ExactSolutionFunction<dim,nstate,real>()
+{
+    t = 0;//time_compare;
+}
+
+template <int dim, int nstate, typename real>
+inline real ExactSolutionFunction_Zero<dim,nstate,real>
+::value(const dealii::Point<dim,real> &/*point*/, const unsigned int /*istate*/) const
+{
+    //NEED TO UPDATE THIS
+    //(void) point;
+    real value = 0;
+    return value;
+}
+
+// ========================================================
+// 1D SINE -- Exact solution for advection_explicit_time_study
+// ========================================================
+template <int dim, int nstate, typename real>
+ExactSolutionFunction_1DSine<dim,nstate,real>
+::ExactSolutionFunction_1DSine (double time_compare)
+        : ExactSolutionFunction<dim,nstate,real>()
+{
+    t = time_compare;
+}
+
+template <int dim, int nstate, typename real>
+inline real ExactSolutionFunction_1DSine<dim,nstate,real>
+::value(const dealii::Point<dim,real> &point, const unsigned int /*istate*/) const
+{
+    double x_adv_speed = 1.0;
+
+    real value = 0;
+    real pi = dealii::numbers::PI;
+    if(point[0] >= 0.0 && point[0] <= 2.0){
+        value = sin(2*pi*(point[0] - x_adv_speed * t)/2.0);
+    }
+    return value;
+}
+
+
+//=========================================================
+// FLOW SOLVER -- Exact Solution Base Class + Factory
+//=========================================================
+template <int dim, int nstate, typename real>
+ExactSolutionFunction<dim,nstate,real>
+::ExactSolutionFunction ()
+    : dealii::Function<dim,real>(nstate)//,0.0) // 0.0 denotes initial time (t=0)
+{
+    //time_compare +=0;
+}
+
+template <int dim, int nstate, typename real>
+std::shared_ptr<ExactSolutionFunction<dim, nstate, real>>
+ExactSolutionFactory<dim,nstate, real>::create_ExactSolutionFunction(
+    Parameters::AllParameters const *const param)
```

Comment:
Don't pass AllParameters if FlowSolverParam is enough.

Reply:
You're right, I'm only using parameters in the flow solver section. How do I pass only FlowSolverParam?

Reply:
The function signature would be `create_ExactSolutionFunction(const FlowSolverParam& flow_solver_parameters)`

And the calling function would call it like `create_ExactSolutionFunction(all_parameters->flow_solver_param)`

Reply:
Thanks, I've changed to only pass FlowSolverParam.